ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OGridSearch.scala

```diff
@@ -205,16 +192,28 @@ class H2OGridSearch(override val uid: String)
   def getGridModelsParams(): DataFrame = {
     ensureGridSearchIsFitted()
     val hyperParamNames = getHyperParameters().keySet().asScala.toSeq
+    val h2oToSwParamMap =
+      getAlgo().getSWtoH2OParamNameMap().map(_.swap)
     val rowValues = gridModels.zip(gridModels.map(_.uid)).map {
       case (model, id) =>
-        val outputParams = model.getTrainingParams().filter { case (key, _) => hyperParamNames.contains("_" + key) }
+        val outputParams =
+          model.getTrainingParams().filter {
+            case (key, _) =>
+              !IgnoredParameters.all.contains(key) && hyperParamNames.contains(h2oToSwParamMap(key))
```

Comment:
Note: currently, we ignore parameters also in `IgnoredParamters.implementedInParent` such as `distribution` etc.

Parameters in this array are not gridable anyways, so not actual issue, but after merging https://github.com/h2oai/sparkling-water/pull/2129 only parameters in this array left are training and validation frame, so all good